src/main/scala/com/microsoft/hyperspace/Hyperspace.scala

```diff
@@ -40,7 +40,9 @@ class Hyperspace(spark: SparkSession) {
    * @param indexConfig the configuration of index to be created.
    */
   def createIndex(df: DataFrame, indexConfig: IndexConfig): Unit = {
-    indexManager.create(df, indexConfig)
+    withHyperspaceDisabled(spark) {
```

Comment:
NOTE: enabled hyperspace during the index creation can be a problem because of the following code in HyperspaceOneRule
```scala
 val indexManager = Hyperspace
      .getContext(spark)
      .indexCollectionManager
    val allIndexes = indexManager.getIndexes(Seq(Constants.States.ACTIVE))
```

Reply:
Sorry, but I still don't understand. 

1. Shouldn't `indexManager.getIndexes` return the same answer in all cases?
2. How does the index creation work if the context is not set through `getContext`? Where will that new index be created, in which location?

Reply:
Ok.. I've analyzed the exact cause. `getIndexes` in HyperspaceOneRule causes caching indexes while createIndex.
So cached `indexes` doesn't have the newly created one. 
```
override def create(df: DataFrame, indexConfig: IndexConfig): Unit = {
    clearCache()
    super.create(df, indexConfig)
    clearCache() // this can be another solution
  }
```

But I think it's also good to disable Hyperspace during creating or refreshing indexes, to avoid any unintended plan change.